src/utils.js

```diff
@@ -66,3 +66,49 @@ export const objectQueryString = obj => {
     .map(([k, v]) => `${k}:${serializeAttribute(v)}`)
     .join(';');
 };
+
+/**
+   Compute longest common path prefix for an array of path parts separated by "/".
+   The input array must be sorted lexically.
+*/
+const longestPathPrefix = (sortedPathParts, result = '') => {
+  if (sortedPathParts.length === 0) {
+    return '';
+  }
+  if (sortedPathParts.length === 1) {
+    const comps = sortedPathParts[0];
+    if (comps.length === 1) {
+      return '';
+    }
+    const path = comps.slice(0, comps.length - 1).join('/');
+    return `${path}/`;
+  }
+
+  const { length: l, 0: first, [l - 1]: last } = sortedPathParts;
```

Comment:
Whou I did not know you can do destructions like this. TIL!

Reply:
Just learned it also while I was doing this 😀

Reply:
I love that typo / autocorrect :)